create MapPinIcon (DEV-1683)

tglaz opened the pull request:
### Create Map Pin icon for react-native Map UI
DEV-1683


Note: 
* demo available on this PR: go to **menu > settings**

<img width="347" alt="image" src="https://github.com/user-attachments/assets/31683153-cba6-4499-923e-93806b5f1c20" />

shiozakiyuko reviewed (commented):
Hi @tglaz 

I noticed a couple of things.

1. I see 44+ in size Small and Mid pins. + shouldn't be there since + is for 99+ only which indicates more than 100. 
2. Since we decided to use the empty pin for a single pin, can we use these instead (screen shot)? I can add this to the ticket if you can replace it. Let me know. 
3. On Android, the pins are flashing. Please see it on the Android device.
4. I don't see AC 3biii, and 3biv.


<img src="https://github.com/user-attachments/assets/f30424dc-92f6-4272-b670-848380605b04" width="300" />

https://github.com/user-attachments/assets/b3875ee9-0bb7-4c35-ac53-b488543e4cd6

tglaz commented:
@shiozakiyuko 

> I see 44+ in size Small and Mid pins. + shouldn't be there since + is for 99+ only which indicates more than 100.

- That should be fine as the icon itself does not care. It will depend on specific implementation when we show the `+`. But I updated the example.

> Since we decided to use the empty pin for a single pin, can we use these instead (screen shot)? I can add this to the ticket if you can replace it. Let me know.

- Nope. That's a new ticket.

> On Android, the pins are flashing. Please see it on the Android device.
- I made an update that may fix it. But if it doesn't, I'd just ignore. The icon itself doesn't flicker, and this PR does not actually implement the icons anywhere, so when they're actually added to a map and flashing then we can address it. 

> I don't see AC 3biii, and 3biv.
- I added green ones to the example. But the icon can accept any border, fill and/or text color

tglaz commented on libs/expo/shared/icons/src/lib/components/MapPinIcon/MapPinIcon.tsx:
```diff
@@ -0,0 +1,88 @@
+import { Colors } from '@monorepo/expo/shared/static';
+import { memo } from 'react';
+import isEqual from 'react-fast-compare';
```
not sure the default `memo` takes care of comparing objects

shiozakiyuko reviewed (commented):
Hi @tglaz,

Thanks for the updates! However, some of the numbers in the pins are not correct.

They should be: 
- Small size: 1-9
- Med size: 10-29
- Large size: 30-98
- XLarge: 99+

<img src="https://github.com/user-attachments/assets/5c9f2482-28de-48df-85df-f0c3b2534cbb" width="400" />

Thanks.

tglaz commented:
@shiozakiyuko  This PR just builds the icon and does not implement logic that is not specific to the icon.
The icon uses styles that are determined by `size` and `variant` properties.
But technically you can render any number or text in it. The logic of using `medium` icon for numbers `10-29` is specific to the Locations feature, but not relevant to the icon itself.

shiozakiyuko commented:
Thanks @tglaz . I see the updates with green pins now (no flashing). So I assume the word "Last Seen" will come with the logic as well, correct? I can approve it now.
